src/AutoMapper/Execution/DelegateFactory.cs

```diff
@@ -145,10 +144,8 @@ private static LateBoundCtor GenerateConstructor(Type type)
                     .Select(p => Expression.Constant(p.DefaultValue, p.ParameterType)).ToArray();
 
                 //create the ctor expression
-                var ctorExpression =
-                    Expression.Lambda<LateBoundCtor>(Expression.Convert(Expression.New(ctorWithOptionalArgs, args),
```

Comment:
No reason to cast to object because allready is one.  Very slight performance gain